### PR TITLE
check image os/arch on push

### PR DESCRIPTION
**Pull request by rolyatmax**
This change has the CLI checking image os and arch before pushing. If the os/arch is not `linux/amd64`, it warns the user there may be issues spawning and they should try rebuilding with the `--platform=linux/amd64` flag. The user is then given the option to ignore the warning and push anyway.

Also, this PR fixes an issue where `404`s were treated as AuthenticationErrors.

Screenshot:

Screen Shot 2022-10-31 at 4 35 46 PM

### CODE REVIEW

File: src/container-manager.ts [hunk 1]
```diff
@@ -65,6 +77,13 @@ export class DockerContainerManager implements ContainerManager {
 export class PodmanContainerManager implements ContainerManager {
   constructor(private command: string) {}
 
+  getImagePlatform(imageName: string): ImagePlatformResult {
+    const getPlatform = spawnSync(this.command, ['image', 'inspect', '--format', '{{.Os}} {{.Architecture}}', imageName])
```
Comment (rolyatmax) [inline src/container-manager.ts:81]: Haven't tested podman, but [the docs](https://docs.podman.io/en/latest/markdown/podman-inspect.1.html) indicate that the `inspect` command behaves the same as it does in docker.

**Comment by paulgb** [comment]
Love it!

File: src/container-manager.ts (hunk 1 above)
Comment (paulgb) [reply src/container-manager.ts:81]: I kinda want to rip out support for podman, I don't think anybody using jamsocket uses it.

**Review by paulgb — APPROVED** [review]

File: src/container-manager.ts (hunk 1 above)
Comment (rolyatmax) [reply src/container-manager.ts:81]: Yeah, me too.

Comment (rolyatmax) [reply src/container-manager.ts:81]: https://github.com/drifting-in-space/jamsocket-cli/pull/33

**Pull request merged**